test/allow-extended-operators.test.js

```diff
@@ -51,12 +36,193 @@ describe('Model.settings.allowExtendedOperators', () => {
+    context('DAO.find()', () => {
+      it('converts extended operators to string value by default', () => {
+        const TestModel = createTestModel();
+        return TestModel.find(extendedQuery()).then((results) => {
+          should(results[0].value).eql('[object Object]');
+        });
+      });
+
+      it('preserves extended operators with allowExtendedOperators set', () => {
+        const TestModel = createTestModel({allowExtendedOperators: true});
+        return TestModel.find(extendedQuery()).then((results) => {
+          should(results[0].value).eql({$exists: true});
+        });
+      });
+
+      function extendedQuery() {
+        // datasource modifies the query,
+        // we have to build a new object for each test
+        return {where: {value: {$exists: true}}};
+      }
+    });
+
+    context('DAO.updateAttributes()', () => {
+      it('`options.allowExtendedOperators` override connector settings - disable strict check', () => {
+        const TestModel = createTestModel({allowExtendedOperators: false}, {strict: true});
+        return TestModel.create({value: 'test'}).then(function(instance) {
+          let updateData = {$set: {value: 'changed'}};
+          return instance.updateAttributes(updateData, {allowExtendedOperators: true})
+            .then(function() {
+              should(TestModel.lastPersistedData).eql(updateData);
+            });
+        });
+      });
+
+      it('`options.allowExtendedOperators` override connector settings - enable strict check', () => {
+        const TestModel = createTestModel({allowExtendedOperators: true}, {strict: true});
+        return TestModel.create({value: 'test'}).then(function(inst) {
+          return inst.updateAttributes({$set: {value: 'changed'}}, {allowExtendedOperators: false})
+            .catch(function(err) {
+              should.exist(err);
+              should(err.name).equal('ValidationError');
+            });
+        });
+      });
+
+      it('`Model.settings.allowExtendedOperators` override connector settings - ' +
+        'disable strict check', () => {
+        const TestModel = createTestModel({allowExtendedOperators: false},
+          {strict: true, allowExtendedOperators: true});
+        return TestModel.create({value: 'test'}).then(function(instance) {
+          let updateData = {$set: {value: 'changed'}};
+          return instance.updateAttributes(updateData).then(function() {
+            should(TestModel.lastPersistedData).eql(updateData);
+          });
+        });
+      });
+
+      it('`Model.settings.allowExtendedOperators` override connector settings - ' +
+        'enable strict check', () => {
+        const TestModel = createTestModel({allowExtendedOperators: true},
+          {strict: true, allowExtendedOperators: false});
+        return TestModel.create({value: 'test'}).then(function(inst) {
+          return inst.updateAttributes({$set: {value: 'changed'}}).catch(function(err) {
```

Comment:
Ditto, you need to convert the case when `updateAttributes` passed into a promise rejection.

Alternatively, perhaps don't set the strict mode and let `updateAttributes` to set the value, similarly to what you do with `find` [here](https://github.com/strongloop/loopback-datasource-juggler/pull/979/files#diff-2efdf289a9ad4ebbfcfa83d9a4c3152aR61)?